File: exo8/exo8.rb
class Exo8
    def self.triangle(hauteur, char)
        for i in 0..hauteur-2
            for j in 0..i+1
                print char[0] + " "
            end
            print "\n"
        end
    end

    def self.main()
        puts "Veuillez entrer la hauteur du triangle : "
        h = Integer(gets.chomp)
        
        puts "Veuillez entrer le caractere a afficher :  "
        c = gets.chomp
    
        triangle(h,c)
    end
end
        
Exo8.main()

File: exo8/exo8.java
import java.util.*;

public class exo8 {
    private static void triangle(int hauteur, char c) {
        for (int i = 0; i < hauteur; i++) {
            for (int j = 0; j <= i; j++) {
                System.out.print(c + " ");
            }
            System.out.print("\n");
        }
    }
    
    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        
        int h = 5;
        System.out.println("Veuillez entrer la hauteur du triangle : \n");
        if (reader.hasNextInt()) {
            h = reader.nextInt();
        }
        
        char c = ' ';
        System.out.println("Veuillez entrer le caractère à afficher :  \n");
        if (reader.hasNext()) {
            c = reader.next().charAt(0);
        }
        
        reader.close();
        System.out.println("\n");
        
        triangle(h, c);
    }
}